extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/JdbcMetaStoreManagerFactory.java

```diff
@@ -106,12 +111,21 @@ private void initializeForRealm(
   }
 
   private DatasourceOperations getDatasourceOperations(boolean isBootstrap) {
-    DatasourceOperations databaseOperations = new DatasourceOperations(dataSource);
+    // TODO: remove when multiple dataSources are supported.
+    if (dataSources.size() > 1) {
+      throw new IllegalStateException("More than one dataSources configured");
+    }
+    DatasourceOperations databaseOperations = new DatasourceOperations(dataSources.getFirst());
     if (isBootstrap) {
-      // TODO: see if we need to take script from Quarkus or can we just
-      // use the script committed in the repo.
       try {
-        databaseOperations.executeScript("scripts/postgres/schema-v1-postgres.sql");
+        DatabaseType databaseType;
+        try (Connection connection = dataSources.getFirst().getConnection()) {
+          String productName =
+              connection.getMetaData().getDatabaseProductName().toLowerCase(Locale.ROOT);
+          databaseType = DatabaseType.fromDisplayName(productName);
+        }
+        databaseOperations.executeScript(
+            String.format("%s/schema-v1.sql", databaseType.getDisplayName()));
```

Comment:
nit: It'd recommend to make `databaseType.getInitScriptResource()` (or something like that) and return the full path.

Reply:
Now that `DatabaseType.getInitScriptResource()` is available, should we use it here?